[release-2.4.0 < T0835-MG] Correct inconsistencies w.r.t. sync replication (#502)
memgraph/docs: docs/how-to-guides/replication.md

   REPLICA instance 1 is called REP1, its replication mode is SYNC, and it is
   located at IP address `172.17.0.3.` with port `10000`.

>    When the REPLICA instance is running in SYNC mode, after committing a transaction

Reviewer: ```suggestion
   When the REPLICA instance is running in SYNC mode, once the MAIN instance commits a transaction, it will
```

>    When the REPLICA instance is running in SYNC mode, once the MAIN instance commits a transaction, it will

Reviewer: ```suggestion
   Once the MAIN instance commits a transaction, it will
```

Reviewer: I think this is not necessary as we mention "until all REPLICA instances running in the SYNC mode
   confirm they have received the same transaction." later in the sentence

>    the MAIN instance will communicate the changes to all REPLICA instances running 

Reviewer: ```suggestion
   communicate the changes to all REPLICA instances running 
```

>    in SYNC mode and wait until it receives a response or that a timeout is reached. <br/>

Reviewer: what kind of response does it need to receive?

Author: A confirmation that the replica received the changes and applied them.
Perhaps I can re-formulate?

>    in SYNC mode and wait until it receives a response or that a timeout is reached. <br/>

Reviewer: ```suggestion
   in SYNC mode and wait until it receives a response that the changes have been applied to the REPLICAs or that a timeout has been reached. <br/>
```

>    In case of timeout, it will return an error to the user indicating that he should check the 

Reviewer: ```suggestion
   In the case of a timeout, the MAIN instance will return an error to the user proposing a check of the 
```

>    status of the REPLICAS as there might be a network/hardware issue.

Reviewer: REPLICAs' statuses as there might be network or hardware issues.

Author: Done

>    status of the REPLICAs' statuses as there might be network or hardware issues.

Reviewer: ```suggestion
   REPLICAs' statuses as there might be network or hardware issues.
```
